src/components/lab/Thermometer.js

```diff
@@ -0,0 +1,189 @@
+/* eslint-disable react/prop-types */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Shape, Group, Text, Rect } from 'react-konva';
+import {
+  THERMOMETER_RADIUS,
+  THERMOMETER_HEIGHT,
+  THERMOMETER_WIDTH,
+  THERMOMETER_POSITION_Y,
+  THERMOMETER_POSITION_X,
+  GRADUATION_HEIGHT,
+  GRADUATION_FONT_SIZE,
+  GRADUATION_PADDING_LEFT,
+  GRADUATION_WIDTH,
+  GRADUATION_LINE_HEIGHT,
+  THERMOMETER_COLOR,
+  THERMOMETER_STROKE_WIDTH,
+  THERMOMETER_STROKE_COLOR,
+  GRADUATION_MAX_NUMBER_TICKS,
+} from '../../config/constants';
+
+const ThermometerShape = ({
+  stroke,
+  strokeWidth,
+  fill,
+  height = THERMOMETER_HEIGHT,
+}) => {
+  // use trigonometry to get the exact angle matching
+  // the circle and the rectangle
```

Comment:
(1) `// the circle and the rectangle` is not a stand-alone comment: if it's the continuation of line 28, there is space to put the comment on one line,
(2) maybe clarify the comment a bit, I think 'use the trigonometry to get the exact angle' is good, but e.g. add 'this angle is used to...' (do so and so in the shape's drawing)